text/0000-rustdoc-types-maintainers.md

```diff
@@ -0,0 +1,132 @@
+
+Additionally, if an update to `rustdoc-json-types` happens while I'm away from a computer for an extended period of time, there will be a delay in this update being published on crates.io. This happened with format_version 29, which was merged on [April 8th](https://github.com/rust-lang/rust/commit/537aab7a2e7fe9cdf50b5ff18485e0793cd8db62),
+but was only published to crates.io on
+[April 19th](https://github.com/aDotInTheVoid/rustdoc-types/commit/ad92b911488dd42681e3dc7e496f777f556a94f6), due to personal reasons.
+[This almost happened previously](https://github.com/aDotInTheVoid/rustdoc-types/issues/25), but was avoided due to the bors queue being quiet at the time.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This involves:
+
+1. Moving the [github.com/aDotInTheVoid/rustdoc-types](https://github.com/aDotInTheVoid/rustdoc-types/) repo to the `rust-lang` organization, and make `rust-lang/rustdoc` maintainers/owners.
+2. Move ownership of `rustdoc-types` on crates.io to the rustdoc team.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+`rustdoc-types` is a republishing of the in-tree [`rustdoc-json-types`](https://github.com/rust-lang/rust/tree/b8536c1aa1973dd2438841815b1eeec129480e45/src/rustdoc-json-types) crate. `rustdoc-json-types` is a dependency of `librustdoc`, and is the canonical source of truth for the rustdoc-json output format. Changes to the format are made as a PR to `rust-lang/rust`, and will modify `src/rustdoc-json-types`, `src/librustdoc/json` and `tests/rustdoc-json`. None of this will change.
+
+Republishing `rustdoc-json-types` as `rustdoc-types` is done with [a script](https://github.com/aDotInTheVoid/rustdoc-types/blob/17cbe9f8f07de954261dbb9536c394381770de7b/update.sh) so that it is as low maintenance as possible. This also ensures that all format/documentation changes happen in the rust-lang/rust repo, and go through the normal review process there.
+
+The update/publishing process will be moved to T-rustdoc. In the medium term, I (`@aDotInTheVoid`) will still do it, but
+- In an official capacity
+- With bus factor for when I stop.
+
+We (T-rustdoc) will continue to publish a new version of the `rustdoc-types` crate
+every time the upstream implementation changes, and these will be versioned with
+normal SemVer. Changes to rustdoc-json in `rust-lang/rust` will not be accepted
+if they would make it not possible to publish `rustdoc-types`.
+
+## Actual Mechanics of the move
+
+### GitHub
+
+GitHub has a [list of requirements](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository) for transferring repositories. T-infra will handle the permissions of moving the repository into the rust-lang GitHub organization.
+
+At the end of this we should have a crate in the rust-lang GitHub org with T-rustdoc as contributors, and T-infra as owners.
```

Comment:
I'm not sure what this is trying to say here. "crate" is in crates.io, not GitHub typically. I also don't see why "T-infra as owners" means -- but it seems pretty surprising for T-infra to be an owner. Do we mean the rust-lang-owner account on crates.io?

The other sections don't seem to elaborate on that.

I at least am not comfortable with infra being an owner for this crate (it feels very much out of our general scope as a team)

Reply:
> I'm not sure what this is trying to say here. "crate" is in crates.io, not GitHub typically.

Gah sorry, this is my bad writing. I mean the [`aDotInTheVoid/rustdoc-types`](https://github.com/aDotInTheVoid/rustdoc-types/) repo.

> I at least am not comfortable with infra being an owner for this crate or accompanying repository: it feels very much out of our general scope as a team, and very much "just" a rustdoc concern.

Agreed. I meant this in terms of GitHub permissions, not who'se maintaining the crate/repo. If the repo admin permissions should instead go to rust-lang-owner (instead of the infra team or even the rustdoc team) I'm happy to do that.

Reply:
Maybe I'm confused, but aren't we moving that repo into the rust-lang org? If so, then permissions are granted via the team repo and should only consist of teams that need access (in this case, presumably rustdoc).

Infra has access if needed through the org admins, but that is true for all repos in the org so generally doesn't need specific enumeration.

Reply:
In any case, I'm happy with the intent now that I understand it :)

Reply:
> Maybe I'm confused, but aren't we moving that repo into the rust-lang org?

Yes

> If so, then permissions are granted via the team repo and should only consist of teams that need access.

Ah. That makes things even simpler.